lesson16/main.js

```diff
@@ -128,28 +141,59 @@ function createNewIconContainer({ date }) {
   return newIconContainer;
 }
 
-function createCommentIconContainer({ comment }) {
-  const comments = comment;
+function createCommentIconContainer(id, comments) {
   const commentIconContainer = document.createElement("div");
-  commentIconContainer.classList.add("comment-container");
-  if (comments.length > 0) {
-    const commentIcon = document.createElement("img");
-    commentIcon.classList.add("comment-icon");
-    commentIcon.src = "./img/comment.png";
-    commentIcon.width = "14";
-    commentIcon.height = "14";
-    commentIconContainer.appendChild(commentIcon);
-
-    const numOfComments = document.createElement("div");
-    numOfComments.classList.add("comment-num");
-    numOfComments.textContent = comments.length;
-    numOfComments.alt = "コメント数";
-    commentIconContainer.appendChild(numOfComments);
-  }
+  commentIconContainer.classList.add("comment-icon-container");
+  const commentIcon = document.createElement("img");
+  commentIcon.classList.add("comment-icon");
+  commentIcon.src = "./img/comment.png";
+  commentIcon.width = "14";
+  commentIcon.height = "14";
+  commentIcon.dataset.id = id;
+  commentIconContainer.appendChild(commentIcon);
+
+  const numOfComments = document.createElement("div");
+  numOfComments.classList.add("comment-num");
+  numOfComments.textContent = comments.length;
+  numOfComments.alt = "コメント数";
+  numOfComments.dataset.id = id;
+  commentIconContainer.appendChild(numOfComments);
+
+  addClickEventChangeElement(commentIconContainer, false, commentArea);
+
   return commentIconContainer;
 }
 
-function createThumbnail({ image }) {
+function createComments(id, comments) {
+  const ArticleCommentsContainer = document.createElement("div");
+  ArticleCommentsContainer.id = id;
+  ArticleCommentsContainer.classList.add("comment-container");
+  comments.forEach((comment) => {
+    const Comment = createComment(comment);

```

Comment:
IMO
変数名はパスカルケースで統一しましょう
```suggestion
  const articleCommentsContainer = document.createElement("div");
  articleCommentsContainer.id = id;
  articleCommentsContainer.classList.add("comment-container");
  comments.forEach((comment) => {
    const comment = createComment(comment);

```

Reply:
sorry...
dfff9f8b77223d261a4672e668274c8fafa7cab7